bundles/org.eclipse.ui.workbench/Eclipse UI/org/eclipse/ui/dialogs/SearchPattern.java

```diff
@@ -39,21 +39,30 @@ public class SearchPattern {
 
 	/**
 	 * Match rule: The search pattern is a prefix of the search result.
+	 * <p>
+	 * This is the <em>default</em> match rule if no other applies.
 	 */
 	public static final int RULE_PREFIX_MATCH = 0x0001;
 
 	/**
 	 * Match rule: The search pattern contains one or more wild cards ('*' or '?').
 	 * A '*' wild-card can replace 0 or more characters in the search result. A '?'
 	 * wild-card replaces exactly 1 character in the search result.
+	 * <p>
+	 * Unless the pattern ends with ' ' or '>', '*' is automatically added to the
+	 * end of the pattern.
+	 * <p>
+	 * When {@link #RULE_SUBSTRING_MATCH} is in effect, then '*' is automatically
+	 * added to the start of the pattern.
```

Comment:
Well, I see that talking about concrete characters is kind of "internal technical decision", yet I think it also currently fills the gap of describing how exactly this flag behaves when talking about prefix and suffix searches. So do you think this description should be removed altogether (as discussed in the other, related thread), or maybe just reformulating it in a general way (i. e. not mentioning concrete characters used) would help here?

Reply:
The lest we specify, the easier maintenance is. Something that was specified to work in a particular way cannot easily be changed later to work differently. So let's specify the purpose of the flag, but not the implementation.

Reply:
OK, I've reformulated the new paragraphs, so they don't describe what exactly happens with the "internal" pattern. I hope you'll find that OK (or suggest to move it to the updated `matches()` javadoc).

At the same time I have realized we are getting into a little bit "schizophrenic" situation here, because the resulting "internal" pattern is historically part of the public API of this class, yet it was never actually documented what it holds. So it is a little bit challenging stuff. My original intention was just to enhance the original javadocs as necessary for this feature, but as it turns out, a little bit bigger changes are necessary to get a coherent result. Hopefully at least, I haven't made the docs worse... ;)